gps/solver.go

```diff
@@ -105,12 +105,9 @@ type SolveParameters struct {
 	// typical case.
 	Downgrade bool
 
-	// Trace controls whether the solver will generate informative trace output
-	// as it moves through the solving process.
-	Trace bool
```

Comment:
Yep, I know that it looks like it amounts to a double non-nil check, and it may actually be that (I vaguely recall, from when I wrote it a year ago, there being an actual reason for having the extra bool). But that needs its own PR - as you said, the issue is about muddying the water.